modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -4295,6 +4303,15 @@ else if (node.clientRouterNodeId() == null &&
             }
         }
 
+        /**
+         * @param node Joined node.
+         * @return
+         */
+        private boolean isDiscoDataPaketCompressed(TcpDiscoveryNode node) {
+            return allNodesSupport(IgniteFeatures.DATA_PACKET_COMPRESSION)
+                && IgniteFeatures.nodeSupports(node, IgniteFeatures.DATA_PACKET_COMPRESSION);
```

Comment:
I think IgniteFeatures.nodeSupports(node, IgniteFeatures.DATA_PACKET_COMPRESSION) should be first in expression cause it's faster to evaluate it first.